packages/@okta/vuepress-site/docs/guides/event-hook-ngrok/main/index.md

```diff
@@ -0,0 +1,159 @@
+
+### Verify the Event Hook
+
+You must verify the Event Hook to prove that your external service controls the endpoint. See [One-Time Verification Request](/docs/concepts/event-hooks/#one-time-verification-request) for further information on this process.
+
+To complete the one-time verification of the Event Hook:
+
+- After creating the Event Hook, and if your ngrok session and local application are ready to handle the request, click **Verify** to complete the one-time verification step.
+
+or
+
+- After making sure that your  ngrok session and local application are ready for the external verification call, go to the Event Hooks table, click the **Actions** drop-down menu of any **UNVERIFIED** Event Hook, and select **Verify**.
+
+The Event Hook is now set up with a status of **VERIFIED** and is ready to send Event Hook calls to your external service.
+
+>**Note:** A successful Event Hook verification also indicates your local application is working with the ngrok session! Review the ngrok terminal for details on the first `GET` call to your local application.
+
+## Preview, test, and review the Event Hook
+
+With your local application now exposed externally through an ngrok session, you can preview and test Okta Event Hook calls, as well as review details of the calls using the ngrok dashboard. The Okta org is also set up to call your local application when an event is triggered. In this example, the event is triggered when a user is deactivated in the Okta org.
+
+### Preview
+
+To run a preview call of your Event Hook, sign in to your Okta org as the super admin.
+
+1. In the Admin Console, go to **Workflow** > **Event Hooks**.
+2. Locate the Event Hook you created during the set-up step. In this example, select `Deactivated User Event Hook` or the name you gave the Event Hook.
+3. Click the **Actions** menu for this hook, and select **Preview**.
+4. In the **Configure Event Hook request** section, select an event from the **Event Type** drop-down menu. In this example, there is only one: `User deactivated (user.lifecycle.deactivate)`.
+5. Select a previous recent event (in this case, a user deactivation) from the **System Log Event** drop-down menu. The **Preview & Deliver Event Hook** section populates the JSON body of the Event Hook. If no event is available, the JSON body populates with sample data.
+6. Optionally, click **Edit** to modify the JSON body call. For example, you can change the `target` object's property, `displayName`, to `John Doe`.
+7. Ensure that both your ngrok session and local sample application are running.
+
+    >**Note:** If you start a new ngrok session at any time, make sure to update the Event Hook URL.
+
+8. Click **Deliver Request**. The Event Hook Preview displays the status request as either successful or a failure. Check your local application console. The following message displays if successful:
```

Comment:
This works if there is a previous recent event but if I try to Deliver Request with a dummy event b/c there is no event available, the server fails:
<img width="900" alt="Screen Shot 2021-11-04 at 2 24 18 PM" src="https://user-images.githubusercontent.com/67654866/140422744-90b93916-aeb2-438f-97bc-31cb9b1bc634.png">

Need to make the server.js code resilient enough to handle both a recent User Deactivated event and a dummy JSON event.  You can just subscribe to another event type that you don't have a previous event and test with the dummy event if you want to test it.  The JSON payload of all dummy events are the same, regardless of the event type

Reply:
Yes, this completely threw me, too, after re-running this project after a couple months. I have added a note to Edit the dummy event data and add a target. Would that be enough?

Or .... I was thinking of a new Event Type to profile here, too. I was hoping to find something that could easily be tested; I'll do some more poking around to perhaps change the Event Type and code slightly.

Reply:
Updated a new event type, User Create (user.lifecycle.create) and include a direct instruction to Add a user prior to previewing.